src/main/main.js

```diff
@@ -152,17 +152,22 @@ app.on("ready", () => {
     );
 
     truffleIntegration.on("error", async error => {
-      if (mainWindow) {
-        mainWindow.webContents.send(SET_SYSTEM_ERROR, error);
-      }
-
       if (truffleIntegration) {
         await truffleIntegration.stopWatching();
       }
 
       if (chain.isServerStarted()) {
         // Something wrong happened in the chain, let's try to stop it
+        if (mainWindow) {
```

Comment:
It was possible for the `error` here to cause the UI to render before the UI had all the data it needs to render. This change makes sure the chain is ready before sending the error to the `mainWindow`.